locations/storefinders/amasty_store_locator.py

```diff
@@ -52,3 +56,16 @@ def parse_items(self, items: [dict]):
 
     def parse_item(self, item, location, popup_html):
         yield item
+
+    def storefinder_exists(response: Response) -> bool | Request:
+        # Example: https://www.aussiedisposals.com.au/store-locator/
+        # Example: https://www.lcbo.com/en/stores/
+        if response.xpath('//script[contains(text(), "/amlocator/index/ajax/")]/text()').get():
+            return True
+
+        return False
+
+    def extract_spider_attributes(response: Response) -> dict | Request:
+        return {
+            "allowed_domains": [urlparse(response.url).netloc],
```

Comment:
Are you able to check with the existing `mephisto` spider which uses `start_urls` instead of `allowed_domains`. Is it possible to get a custom URL such as `https://www.mephisto.com/be-fr/amlocator/index/ajax/` from `response.xpath('//script[contains(text(), "/amlocator/index/ajax/")]/text()').get()`?

Reply:
Most appear to do a POST as the first request exactly to the endpoint
ie:
```
    def start_requests(self):
        # The request won't work without the headers supplied below.
        headers = {
            "X-Requested-With": "XMLHttpRequest",
        }
        for domain in self.allowed_domains:
            yield Request(url=f"https://{domain}/amlocator/index/ajax/", method="POST", headers=headers)
```

Reply:
The issue in the case of `mephisto` is the API URL has `/be-fr/` in the path hence `f"https://{domain}/amlocator/index/ajax/"` is not correct for `mephisto`. I wonder whether the API URL can be detected from the HTML/JS content of the store finder page, rather than just assuming it is always `f"https://{domain}/amlocator/index/ajax/"`?

Reply:
Perhaps a new spider attribute would be helpful to define whether the instance needs the `POST` method (that also adds the `X-Requested-With` header) or whether it needs a plain `GET` method instead?

Reply:
(I will have to look at this tonight/later, no dev environment at the moment)

Reply:
https://www.mephisto.com/be-fr/points-de-vente/ - 404

Reply:
I guess we can rip out:
            ajaxCallUrl: "https://www.mephisto.com/be-fr/amlocator/index/ajax/",

Reply:
I just tried https://github.com/alltheplaces/alltheplaces/tree/autospidergen with AmastyStoreLocatorSpider modified to subclass additionally from AutomaticSpiderGenerator, and to have a `storefinder_exists` method always return True. The example of `pipenv run scrapy sf https://www.aussiedisposals.com.au/store-locator/` then works for me in returning a template for a new spider.

Reply:
Weird. This branch is definitely wrong on my local. Let me start over.